eio/tls_eio.ml

```diff
@@ -22,14 +22,17 @@ module Raw = struct
     | End_of_file as ex ->
       t.state <- `Eof;
       raise ex
+    | Eio.Cancel.Cancelled _ as ex ->
+      (* No need to break the flow just because a read got cancelled. *)
+      raise ex
     | exn ->
       (match t.state with
        | `Error _ | `Eof -> ()
        | `Active _ -> t.state <- `Error exn) ;
       raise exn
```

Comment:
Wouldn't it make sense to completely remove this bit instead? I believe this is trapping other exceptions in addition to `Cancelled`, such as 'Poisioned' and intermittently this https://github.com/ocaml/ocaml/blob/trunk/stdlib/effect.ml#L59

Reply:
I assume the idea is that the flow should stop working on any unexpected error. Otherwise, you might fail to read a packet, retry, and then hang (because the sender thinks you already have the data). Possibly we should call `Printexc.get_raw_backtrace` here and store that in `t.state` too. Then you'd know where the original error came from.

But note that the original exception is still raised, with the original backtrace, the first time. So the first error you get should include the correct location in all cases. It's just if you keep trying to use it after an error then future errors don't give the exact location.

Reply:
>But note that the original exception is still raised, with the original backtrace, the first time. So the first error you get should include the correct location in all cases. It's just if you keep trying to use it after an error then future errors don't give the exact location.

Indeed, that's what I expected. However, I was getting other exceptions being reported - as mentioned in the comment above - as being the cause of error. As a user of `tls-eio`, that led me to unproductive debugging sessions since the actual error was quite something else. I have a feeling that the interaction of `Eio.Fiber.any/first`, tls_eio exception handling here  and the like is giving unpredictable exceptions. I was only able to move forward with identifying the correct issue after I removed this catch all exception handling. Removing the catch all exception handling also don't seem to impact any observable tls-eio behaviours, i.e. the tests pass and End_of_file exception is correctly reported to users of tls-eio.